src/arrayMethodSort.js

```diff
@@ -4,8 +4,45 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    let defaultCompare;
+
+    if (typeof compareFunction === 'function') {
+      defaultCompare = compareFunction;
+    } else {
+      defaultCompare = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+        // Порівнюємо лексикографічно, ігноруючи регістр
+
+        if (typeof a === 'string' && typeof b === 'string') {
+          // Перевіряємо ASCII код перших символів
+          // Якщо перший символ має більший ASCII код, він має йти раніше
+          if (strA[0] === strB[0]) {
+            // Якщо перші символи однакові, порівнюємо повністю
+            return strA.localeCompare(strB);
+          } else {
+            return strA[0].charCodeAt(0) - strB[0].charCodeAt(0);
+          }
+        }
```

Comment:
The condition checking if both `a` and `b` are strings is redundant. Since `strA` and `strB` are created by coercing `a` and `b` to strings, you can directly use `localeCompare` for comparison. The `localeCompare` method already handles string comparison correctly, so the additional check and character code comparison are unnecessary.